crates/core/src/client_events/websocket.rs

```diff
@@ -518,17 +533,37 @@ async fn process_client_request(
         *auth_token = Some(AuthToken::from(token.clone()));
     }
 
+    // Add special logging for PUT requests to debug River issue
+    if let ClientRequest::ContractOp(ContractRequest::Put { contract, .. }) = &req {
+        let contract_id = contract.key().to_string();
+        tracing::info!(
+            "WebSocket: Received PUT request from client_id={}, contract_key={}",
+            client_id,
+            contract_id
+        );
+    }
+
     tracing::debug!(req = %req, "received client request");
-    request_sender
+
+    let send_result = request_sender
         .send(ClientConnection::Request {
             client_id,
             req: Box::new(req),
             auth_token: auth_token.clone(),
             attested_contract,
         })
-        .await
-        .map_err(|err| Some(err.into()))?;
-    Ok(None)
+        .await;
+
+    match send_result {
```

Comment:
this is the worst part that we were silently swallowing this. In the future we should have a regression test for this forcing some sort of error (it can be done at this component level by mocking the rest of the interface, e.g. dropping the other end of the channel).

good catch